packages/material-ui/test/integration/Menu.test.js

```diff
@@ -1,403 +1,313 @@
-      assert.strictEqual(menuListFocusTracker, true);
-      assert.deepEqual(tabIndexTracker, [false, false, false]);
+      expect(getAllByRole('menuitem')[1]).to.be.focused;
+      expect(getAllByRole('menuitem')[0]).to.have.property('tabIndex', -1);
+      expect(getAllByRole('menuitem')[1]).to.have.property('tabIndex', 0);
+      expect(getAllByRole('menuitem')[2]).to.have.property('tabIndex', -1);
     });
 
-    it('[variant=selectedMenu] second item selected, no autoFocus', () => {
-      mountTrackingMenu('selectedMenu', { selectedIndex: 1, autoFocus: false });
-      assert.deepEqual(contentAnchorTracker, [false, true, false]);
-      assert.deepEqual(focusTracker, [false, false, false]);
-      assert.strictEqual(menuListFocusTracker, false);
-      assert.deepEqual(tabIndexTracker, [false, true, false]);
+    // no case for menu
+    it('[variant=selectedMenu] second item selected, explicit tabIndex', () => {
+      const { getAllByRole } = render(
+        <OpenMenu variant="selectedMenu">
+          <MenuItem />
+          <MenuItem selected tabIndex={2} />
+          <MenuItem />
+        </OpenMenu>,
+      );
+
+      expect(getAllByRole('menuitem')[1]).to.be.focused;
+      expect(getAllByRole('menuitem')[0]).to.have.property('tabIndex', -1);
+      expect(getAllByRole('menuitem')[1]).to.have.property('tabIndex', 2);
+      expect(getAllByRole('menuitem')[2]).to.have.property('tabIndex', -1);
     });
-  });
 
-  describe('closing', () => {
-    let wrapper;
-    let portalLayer;
+    // no case for menu
+    it('[variant=selectedMenu] second item selected and disabled', () => {
+      const { getAllByRole, getByRole } = render(
+        <OpenMenu variant="selectedMenu">
+          <MenuItem />
+          <MenuItem disabled selected />
+          <MenuItem />
+        </OpenMenu>,
+      );
 
-    beforeEach(() => {
-      wrapper = mount(<SimpleMenu transitionDuration={0} />);
-      wrapper.find('button').simulate('click');
-      portalLayer = document.querySelector('[data-mui-test="Modal"]');
+      expect(getByRole('menu')).to.be.focused;
+      expect(getAllByRole('menuitem')[0]).to.have.property('tabIndex', -1);
+      expect(getAllByRole('menuitem')[1]).to.have.property('tabIndex', -1);
+      expect(getAllByRole('menuitem')[2]).to.have.property('tabIndex', -1);
     });
 
-    it('should close the menu with tab', done => {
-      wrapper.setProps({
-        onExited() {
-          assert.strictEqual(document.getElementById('[data-mui-test="Menu"]'), null);
-          done();
-        },
-      });
-      assert.strictEqual(wrapper.text(), 'selectedIndex: null, open: true');
-      const list = portalLayer.querySelector('ul');
-      TestUtils.Simulate.keyDown(list, {
-        key: 'Tab',
-      });
-      assert.strictEqual(wrapper.text(), 'selectedIndex: null, open: false');
-    });
+    // no case for menu
+    it('[variant=selectedMenu] second item selected, no autoFocus', () => {
```

Comment:
What is an actual use case for this? Not sure we should enable disabling a11y.

Reply:
> What is an actual use case for this?

The main use case for `autoFocus=false` is so that the user doesn't have to fight against our focus management if the user is managing focus outside the library in some different manner.

Reply:
> some different manner.

That is my question: Why do we want to give away focus control over an opening menu? What is the purpose of opening a menu but not focusing it?

Reply:
> What is the purpose of opening a menu but not focusing it?

If the user wants to put focus on a particular menu item (e.g. using the MenuItem `autoFocus` property) other than what the library would do by default. I don't feel strongly about keeping this capability -- it existed before my changes (just with a different name). I'm not certain of what scenario prompted the original introduction of the `disableAutoFocusItem` property.

Reply:
Alright let's leave it in if this behavior existed before.